dbt_server/views.py

```diff
@@ -349,46 +386,59 @@ async def post_invocation(args: PostInvocationRequest):
     return JSONResponse(status_code=200, content={})
 
 
-class GetInvocationRequest(BaseModel):
-    # Unique task id of invocation.
-    task_id: str
+@app.get("/invocations/{task_id}")
+async def get_invocation(task_id: str):
+    """Gets invocation entity by `task_id`."""
+    invocation = (
+        convert_celery_result_to_invocation(
+            AbortableAsyncResult(task_id, app=celery_app)
+        )
+        if _lookup_abortable_async_result(task_id)
+        else get_not_found_invocation(task_id)
+    )
 
+    return JSONResponse(status_code=200, content=invocation.dict(exclude_unset=True))
 
-class GetInvocationResponse(BaseModel):
-    # Unique task id of invocation, same as request.
-    task_id: str
-    # Task state, it's one of celery worker state:
-    # - PENDING: The task is pending to be executed or never been created yet.
-    #   Due to celery backend design, we can't tell if task is pending or never
-    #   been created yet. Definitely there is a workaround, but we don't support
-    #   right now.
-    # - STARTED: The task is started by worker.
-    # - SUCCESS: The task is finished by worker.
-    # - FAILURE: The task is failed.
-    # - ABORTED: The task is aborted by user.
-    state: str
-    # Only exists if state = FAILURE, python exception type name that causes
-    # task is killed, e.g. dbtUsageException, WorkerLostError.
-    exc_type: Optional[str]
-    # Only exists if state = FAILURE, similar to exc_type, it includes error
-    # message of exception.
-    exc_message: Optional[str]
-
-
-@app.get("/invocation")
-async def get_invocation(args: GetInvocationRequest):
-    """Gets task state and other metadata."""
-    # TODO: Implement.
-    return JSONResponse(status_code=200, content={})
 
+class ListInvocationResponse(BaseModel):
+    # List of all invocations.
+    invocations: List[Invocation]
 
-class AbortInvocationRequest(BaseModel):
-    # Unique task id of invocation to be aborted.
-    task_id: str
+
+@app.get("/invocations")
+async def list_invocation():
+    """Gets invocation entity by `task_id`."""
+    return JSONResponse(
+        status_code=200,
+        content=ListInvocationResponse(
+            invocations=[
+                convert_celery_result_to_invocation(
+                    AbortableAsyncResult(task_id, app=celery_app)
+                )
+                for task_id in _list_all_task_ids()
+            ]
+        ).dict(exclude_unset=True),
+    )
 
 
-@app.post("/invocation/abort")
-async def abort_invocation(args: AbortInvocationRequest):
-    """Aborts tasks. Notice it's best effort, task may still finish or fail."""
-    # TODO: Implement.
-    return JSONResponse(status_code=200, content={})
+@app.post("/invocation/{task_id}/abort")
+async def abort_invocation(task_id: str):
```

Comment:
cancel! woooo!

An aborted task is in it's own state right, like "ABORTED"?

Reply:
That's right, PTAL schemas.py Invocation state to see all task state enums.